lib/interface_quda.cpp

```diff
@@ -3641,26 +3587,24 @@ void invertMultiShiftQuda(void **_hp_x, void *_hp_b, QudaInvertParam *param)
   // Create the solution fields filled with zero
   cudaParam.create = QUDA_ZERO_FIELD_CREATE;
 
-  // now check if we need to invalidate the solutionResident vectors
+  // now check if we need to invalidate the solution vectors
   bool invalidate = false;
-  for (auto v : solutionResident) {
-    if (cudaParam.Precision() != v->Precision()) {
+  for (auto &v : solutionResident) {
+    if (cudaParam.Precision() != v.Precision()) {
       invalidate = true;
       break;
     }
   }
+  if (invalidate) solutionResident.clear();
```

Comment:
could that just move up and replace the `invalidate=true` ?

Reply:
a40fac9cc1e9d20b00cdb66bc0ee749d0579e99f